### ISSUE TITLE
Player Skins

### ISSUE THREAD

**Issue by xxFLORII** [WaterdogPE/WaterdogPE#141]
Player skins are shown as steve skin for you since 1.18, 
but all other players can see the correct skin you have selected.

**Comment by AlicanCopur** [comment]
Can confirm, I have the same issue with latest WDPE & PowerNukkit.

**Comment by Alemiz112** [comment]
I can not reproduce this issue. Can you provide more information about your skin please?

**Comment by xxFLORII** [comment]
![xxAROX_Skin_christmas](https://user-images.githubusercontent.com/38558911/147890513-fe175150-3fc8-41df-9536-d0b1e8ecabf6.png)

I use PM4 + WaterdogPE and I use this skin.

I also use this plugin:
https://github.com/BlckqPlugins/SkinFix

**Comment by alvin0319** [comment]
Are you sure you have disabled `Only Allow Trusted Skins` on Settings?

**Comment by irrld** [comment]
I can not reproduce this too. Are you using persona skins? What software are you using?

**Comment by irrld** [comment]
This could be something to do with xuids being not sent with the player list packet.

**Comment by Alemiz112** [comment]
> This could be something to do with xuids being not sent with the player list packet.

XUID is not used internally for skin stuff, UUID is. It more seems like downstream software blocking or having problem with certain skins.

**Comment by xxFLORII** [comment]
> Are you sure you have disabled `Only Allow Trusted Skins` on Settings?

yes, i am sure.

**Comment by supercrafter333** [comment]
I can confirm this bug.

**Comment by AlicanCopur** [comment]
Yeah i finally catched new information.
This problem only happens the proxy which using Tobias's LegacyUUIDSupport plugin 
(https://github.com/TobiasG-DE/LegacyUUIDSupport)
@TobiasG-DE @Alemiz112

**Comment by Alemiz112** [comment]
Ah yes, this probably makes sense. The plugin basically fakes an UUID and sends it down to downstream that will use this UUID, which is unknown to the player, as part of PlayerListPacket, which causes the bug.
I would recommend not to use the plugin at all as it is hack. But if you can' get rid of it easily, than you would need to implement PlayerListPacket handling and overwriting modified player UUID to the correct one.

**Comment by irrld** [comment]
Oh sorry this was a WDPE plugin lol.

**Comment by irrld** [comment]
@AlicanCopur can you test with this?
https://github.com/irrld/LegacyUUIDSupport
There is a compiled version too if you want https://www.mediafire.com/file/uo5q0yv4t2n1xk1/LegacyUUIDSupport-1.0-SNAPSHOT.jar/file

**Comment by AlicanCopur** [comment]
> @AlicanCopur can you test with this? https://github.com/irrld/LegacyUUIDSupport There is a compiled version too if you want https://www.mediafire.com/file/uo5q0yv4t2n1xk1/LegacyUUIDSupport-1.0-SNAPSHOT.jar/file

Yeah, that works fine. Hope this will be useful for everyone who have that problem..

**Comment by irrld** [comment]
Okay nice, i'll open pull request Tobias will look at it.
